src/poetry/utils/env/base_env.py

```diff
@@ -242,6 +244,41 @@ def paths(self) -> dict[str, str]:
 
         return self._paths
 
+    @property
+    def installer_scheme_dict(self) -> dict[str, str]:
+        """
+        This property exists to allow cases where system environment paths are not writable and
+        user site is enabled. This enables us to ensure packages (wheels) are correctly installed
+        into directories where the current user can write to.
+        """
+        if self._installer_scheme_dict is None:
+            paths = self.paths.copy()
+
+            if not self.is_venv() and paths.get("userbase"):
+                overrides: dict[str, str] = {}
+                base_path = os.path.commonpath([paths["include"], paths["purelib"]])
+
+                for key in [
+                    "include",
+                    "purelib",
+                    "platlib",
+                    "stdlib",
+                    "platinclude",
+                    "scripts",
+                    "data",
+                ]:
+                    if not is_dir_writable(path=Path(paths[key]), create=True):
+                        overrides[key] = paths[key].replace(
```

Comment:
pretty sure there is a feature request somewhere asking for `--user` type installs, perhaps it makes sense to do that instead, have this be explicit

would it get messy if eg you provide `--user` on `install` but forget it on `update`, or similar?  I dont know.

Reply:
I suspect that would confuse things more from the user perspective. Especially because we are predominantly servicing virtual environments. 

Such a flag would in theory only be used when the user knows they are running in a system environment and does not have write access to the system environment. Hence, I'm not sure how valuable it might be as a cli option. Maybe a config option, I do not know. 

If I understood the issues requesting `--user`, this would simply solve the root issue for them.

Reply:
> Is it ok to end up potentially with a mix of system and user directories here?  This is not something I have looked into but I would have guessed you'd want all or nothing

In practice it shouldn't matter. And in reality it is highly unlikely that parts of it is writable and others not. 

Python will look everywhere by default.

Reply:
hmm.  `pip` checks all directories when before making the decision I think - https://github.com/pypa/pip/blob/f4a543917eac822a3b06ab0fcc4df7c4e37e387a/src/pip/_internal/commands/install.py#L652.  And then once it has made that decision it is fully committed to a user install rather than a site-wide install.

Reply:
We can replicate that I guess.